### PR TITLE
 planner/core: fix the stats of window function

### LINKED ISSUE #9591: Panicked when using window function with normal agg function
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```
CREATE TABLE t1 (id INTEGER, sex CHAR(1));
SELECT sex, AVG(id), MIN(AVG(id)) OVER w min, MAX(AVG(id)) OVER w max FROM t1
    GROUP BY sex HAVING sex='M' OR sex='F' OR sex IS NULL
    WINDOW w AS (ROWS UNBOUNDED PRECEDING) ORDER BY sex DESC;
```

2. What did you expect to see?

Execute successfully.

3. What did you see instead?

TiDB panicked

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

master e7ff050a453c8b8564b30c7c55b993a11c44b1d9

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

fix #9591 

### What is changed and how it works?

Columns involved in window function may not be the same with its child， since there can be a projection elimination and this projection does the column mapping before.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: planner/core/stats.go [hunk 1]
```diff
@@ -341,12 +341,15 @@ func (p *LogicalMaxOneRow) DeriveStats(childStats []*property.StatsInfo) (*prope
...
 func (p *LogicalWindow) DeriveStats(childStats []*property.StatsInfo) (*property.StatsInfo, error) {
 	childProfile := childStats[0]
-	childLen := len(childProfile.Cardinality)
 	p.stats = &property.StatsInfo{
 		RowCount:    childProfile.RowCount,
-		Cardinality: make([]float64, childLen+1),
+		Cardinality: make([]float64, p.schema.Len()),
+	}
+	for i := 0; i < p.schema.Len()-1; i++ {
+		colIdx := p.children[0].Schema().ColumnIndex(p.schema.Columns[i])
```
Comment (winoros) [inline planner/core/stats.go]: Here we directly visit `p.Children`, which i think is a not compatible operation with current cascades. /cc @eurekaka

File: planner/core/stats.go [hunk 2]
```diff
@@ -341,12 +341,15 @@ func (p *LogicalMaxOneRow) DeriveStats(childStats []*property.StatsInfo) (*prope
...
+		p.stats.Cardinality[i] = childProfile.Cardinality[colIdx]
 	}
 	copy(p.stats.Cardinality, childProfile.Cardinality)
```
Comment (eurekaka) [inline planner/core/stats.go]: This `copy` is not needed now?

File: planner/core/stats.go (hunk 1 above)
Comment (eurekaka) [reply planner/core/stats.go]: Yes, cascades planner would panic then. Would this column order change caused by projection elimination impact other codes of window function?

Comment (lamxTyler) [reply planner/core/stats.go]: Yes, seems after the fix the field info is missing:
```sql
MySQL [test]> select avg(b), max(avg(b)) over() from t group by a;
+--------+--------------------+
|        | max(avg(b)) over() |
+--------+--------------------+
| 1.5000 |             1.5000 |
| 1.5000 |             1.5000 |
+--------+--------------------+
```

Comment (winoros) [reply planner/core/stats.go]: @eurekaka @lamxTyler The missing field name is fixed in https://github.com/pingcap/tidb/pull/9644.

**Comment by zz-jason** [comment]
@winoros Please merge master && resolve conflicts.

**Pull request merged**